subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesKotlinDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,2252 @@
+                    val testCompileClasspathFileNames = configurations.getByName("testCompileClasspath").files.map { it.name }
+                    val testRuntimeClasspathFileNames = configurations.getByName("testRuntimeClasspath").files.map { it.name }
+
+                    assert(testCompileClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "servlet-api-3.0-alpha-1.jar")))
+                    assert(!testCompileClasspathFileNames.contains("mysql-connector-java-8.0.26.jar")) { "runtimeOnly dependency" }
+                    assert(testRuntimeClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "mysql-connector-java-8.0.26.jar")))
+                    assert(!testRuntimeClasspathFileNames.contains("servlet-api-3.0-alpha-1.jar")) { "compileOnly dependency" }
+                }
+            }
+             """
+
+        expect:
+        succeeds 'checkConfiguration'
+    }
+
+    def 'can add dependency providers which provide GAVs to the implementation, compileOnly and runtimeOnly configurations of a suite'() {
+        given :
+        buildKotlinFile << """
+            plugins {
+              `java-library`
+            }
+
+            ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+            val commonsLang = project.provider { "org.apache.commons:commons-lang3:3.11" }
+            val servletApi = project.provider { "javax.servlet:servlet-api:3.0-alpha-1" }
+            val mysql = project.provider { "mysql:mysql-connector-java:8.0.26" }
+
+            testing {
+                suites {
+                    val test by getting(JvmTestSuite::class) {
+                        dependencies {
+                            implementation(commonsLang)
+                            compileOnly(servletApi)
+                            runtimeOnly(mysql)
+                        }
+                    }
+                }
+            }
+
+            tasks.register("checkConfiguration") {
+                dependsOn("test")
+                doLast {
+                    val testCompileClasspathFileNames = configurations.getByName("testCompileClasspath").files.map { it.name }
+                    val testRuntimeClasspathFileNames = configurations.getByName("testRuntimeClasspath").files.map { it.name }
+
+                    assert(testCompileClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "servlet-api-3.0-alpha-1.jar")))
+                    assert(!testCompileClasspathFileNames.contains("mysql-connector-java-8.0.26.jar")) { "runtimeOnly dependency" }
+                    assert(testRuntimeClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar", "mysql-connector-java-8.0.26.jar")))
+                    assert(!testRuntimeClasspathFileNames.contains("servlet-api-3.0-alpha-1.jar")) { "compileOnly dependency" }
+                }
+            }
+             """
+
+        expect:
+        succeeds 'checkConfiguration'
+    }
+
+    @Ignore("exclude not yet present in Kotlin DSL")
```

Comment:
I wonder if it wouldn't be more prudent to _not_ ignore these, but instead do the required cast, e.g.
```kotlin
                            implementation(beanUtils) {
                                this as ExternalModuleDependency
                                exclude(group = "commons-collections", module = "commons-collections")
                            }
```
This should work, and should also continue working.